### ISSUE TITLE
OneSignal is starting my application class after closing my app

### ISSUE THREAD

**Issue by Reza-Soltani** [OneSignal/OneSignal-Android-SDK#458]
Hi,

Seems that after closing my app, OneSignal is creating my app's application class for SyncJobService. I know this is not a bug, but I am initializing a lot's of things in onCreate method of my application class.

Leakcanary has a very usefull method, named isInAnaluzerProcess that determines: "is the reason of calling onCreate leakcanary or not"?

You may see it's usage here:
```java
  @Override public void onCreate() {
    super.onCreate();
    if (LeakCanary.isInAnalyzerProcess(this)) {
      // This process is dedicated to LeakCanary for heap analysis.
      // You should not init your app in this process.
      return;
    }
```

Can you provide us a method like this, to prevent initializing the app in OneSignal background process? 

Thanks

**Comment by jkasten2** [comment]
@Reza-Soltani OneSignal does not have a process start reason like LeakCanary does. LeakCanary is a bit different too since it starts it's own process where OneSignal does not.

Depending on what you need to initialize you may want to defer it to when the first Activity is started. This can be done with the `registerActivityLifecycleCallbacks` noted below.
http://www.klaasnotfound.com/2015/08/24/tracking-the-application-lifecycle-on-android/

**Comment by jkasten2** [comment]
I looked into if anything can be done at the time `Application.onCreate` is fired but it doesn't seem possible. Android's doc note that [`Application.onCreate`](https://developer.android.com/reference/android/app/Application.html#onCreate()) is called before any service, activity, etc are created. There isn't a way to get the `Intent` that started the process either from what I could find.

I looked into LeakCanary's source for [`isInAnalyzerProcess`](https://github.com/square/leakcanary/blob/2ae03d85120c2b4b6abcf65cb59cab7221c3f74f/leakcanary-android/src/main/java/com/squareup/leakcanary/LeakCanary.java#L144-L146) and found it runs [`isInServiceProcess`](https://github.com/square/leakcanary/blob/d19c9dc7134a10386ca276595f74f148c028c9d7/leakcanary-android/src/main/java/com/squareup/leakcanary/internal/LeakCanaryInternals.java#L92) which checks for the process name which is why it is possible for this library.

I believe using `registerActivityLifecycleCallbacks` as I noted above will be your best option to only initialize some heavier parts of your app when the first `Activity` is created. However I am open to any suggestions if I am mistaken that our SDK can do to help. Thanks!
